cognitedata/reveal
# refactor: make annotation getter async and extract from texture loading

#### Type of change

![Refactor](https://img.shields.io/badge/Type-Refactor-lightgrey) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-906

## Description :pencil:


- Factor out annotations to not go through cache. The hypothesis is that the annotations will not pose a significant memory footprint, even though they are never expelled from memory in a session
- Make annotation getter async
- Outsource default styling handling to each revision

## How has this been tested? :mag:

- Start Reveal examples
- Open https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=4923307033982497&revisionId=3100284276970593
- Load 360-image set `helideck-site-2-jpeg`
- Enter lower-most 360 image and the one beside it (the latter has one annotation on the ground)
- Observe that annotations are loaded and displayed correctly
- Observe that the annotation default styling (set from the UI) works properly
- Note that clicking an annotation will set a custom appearance overriding the default style, but it should disappear when clicking another annotation




## Test instructions :information_source:




## Checklist :ballot_box_with_check:
